xarray/core/dataset.py

```diff
@@ -1220,6 +1220,26 @@ def sel(self, method=None, tolerance=None, drop=False, **indexers):
         Dataset.isel_points
         DataArray.sel
         """
+        from .dataarray import DataArray
+
+        for k, v in indexers.items():
+            if isinstance(v, tuple):
+                if (k in self.indexes and
+                    (isinstance(self.indexes[k].data, pd.MultiIndex) or
+                     self.indexes[k].dtype == 'object')):
+                    # If array dtype is tuple, we should be carefully check
+                    # whether indexer should be Variable or not.
+                    try:
+                        v_tmp = as_variable(v)
```

Comment:
I think 
```python
ds.sel(x=(('x', ), [0, 1, 2]))
```
should be a good shortcut of 
```python
index = Variable(('x', ), [0, 1, 2])
ds.sel(x=index)
```
but it assumes that *array dtype is not tuple*.
I noticed that it is not the case for `MultiIndex` array, dtype of which is thought tuple ([example](https://github.com/pydata/xarray/blob/master/xarray/tests/test_dataset.py#L1077)).

Although we can automatically detect whether the indexer should be a `Variable` or a `tuple`,
I am wondering it would be error prone.